addons/event/models/event_mail.py

```diff
@@ -74,32 +84,61 @@ class EventMailScheduler(models.Model):
     interval_type = fields.Selection([
         ('after_sub', 'After each registration'),
         ('before_event', 'Before the event'),
-        ('after_event', 'After the event')],
+        ('after_event', 'After the event'),
+        ('stage_update', 'On stage update')],
         string='Trigger ', default="before_event", required=True)
+    trigger_stage_id = fields.Many2one(
+        'event.stage', string='Stage', readonly=False, store=True, copy=True, compute='_compute_trigger_stage_id',
+        help='Used when the interval type is "On stage update". When the event is moved into this stage, the event mail will be triggered.')
+    on_update_stage_date = fields.Datetime('Stage change date', help='Date when the event was moved to the sage')
     template_id = fields.Many2one(
         'mail.template', string='Email Template',
         domain=[('model', '=', 'event.registration')], ondelete='restrict',
         help='This field contains the template of the mail that will be automatically sent')
     scheduled_date = fields.Datetime('Scheduled Sent Mail', compute='_compute_scheduled_date', store=True)
     mail_registration_ids = fields.One2many('event.mail.registration', 'scheduler_id')
     mail_sent = fields.Boolean('Mail Sent on Event', copy=False)
-    done = fields.Boolean('Sent', compute='_compute_done', store=True)
+    done = fields.Boolean('Sent', compute='_compute_done', store=True, copy=False)
+
+    # visual fields
+    state = fields.Selection([('running', 'Running'), ('scheduled', 'Scheduled'), ('sent', 'Sent')], compute='_compute_state')
+    sent_to_message = fields.Char('Sent to message', readonly=True, copy=False)
```

Comment:
@tde-banana-odoo at the beginning, it was unstored, but the counter was wrong with a computed field (because we do not store the number of attendee that receive the mail when the trigger type is "after/before the event") :/